RESTserver/src/main/java/com/softserve/if072/restservice/service/StorageService.java

```diff
@@ -91,14 +98,18 @@ public void update(StorageDTO storageDTO) {
 
         int diff;
         if ((diff = storage.getAmount() - storageDTO.getAmount()) > 0) {
-            addToHistory(storage, diff);
+            addToHistory(storage, diff, Action.USED);
+        } else {
+            addToHistory(storage, -diff, Action.PURCHASED);
         }
         storage.setAmount(storageDTO.getAmount());
         storageDAO.updateAmount(storage);
 
         if (storage.getAmount() <= 1) {
             shoppingListService.insert(new ShoppingList(storage.getUser(), storage.getProduct(), 1));
         }
+
+        messageService.broadcastSpittle("storage updated", userService.getById(storageDTO.getUserId()));
```

Comment:
Please localize message. Could you please specify message to have exect information, like
You just get 1 bottle of oil. 3 more left in storage...